### DISCUSSION TITLE
Support PGP encrypted subjects

### DISCUSSION THREAD

**Discussion by jasperweiss** [simple-login/app#529] [Feature Requests]
Apparently [ProtonMail does support encrypted subjects received externally  ](https://www.reddit.com/r/ProtonMail/comments/obfnjp/subject_is_endtoend_encrypted/). It'd be nice if Simplelogin could encrypt the subject headers as well. 

![afbeelding](https://user-images.githubusercontent.com/29832921/125162235-04a9bf00-e187-11eb-82ac-527a42c5359a.png)

**Comment by nguyenkims** [comment]
As encrypting email subject isn't supported by all PGP softwares, I'd suggest using the "generic subject" feature on SimpleLogin to hide the original subject. This way the original subject is also protected when you reply to a forwarded email.

**Reply by jasperweiss** [reply]
I'm aware of this feature but it makes it very hard to find anything. You end up having to open each email. I've also encountered cases in the past where this broke the email's encoding such that they were rendered as gibberish.

**Reply by nguyenkims** [reply]
Do you know if the encoding issue is still there? A fix has been deployed some times ago. I think we wait for the generic subject to be supported by popular PGP software before it's implemented in SimpleLogin.

**Reply by jasperweiss** [reply]
The specific emails that had the encoding issue do seem to be encoded correctly now. I'd still like to see encrypted subjects though. It's a much more elegant solution than generic subjects as you can actually see (and search for) the subject without opening each email. Not protecting the subject at all means an eavesdropper can more or less tell most of what's inside the email anyway without even being able to read it. Most services even put the most crucial information, like verification codes and order numbers, right inside the subject, making PGP encryption mostly useless.

**Reply by dbingham** [reply]
Another consideration here is that I don't necessarily want to "hide" the subject when replying to a forwarded email.  I want it hidden from my email provider, but I'd love it if the party on the other side of the exchange couldn't even tell that anything was different from any other normal email exchange.  Replacing with a generic subject breaks that expectation and breaks things like threading on some platforms/clients.  If I can keep all of the relevant bits hidden on _my_ side, but let the other side interact with the thread like normal, that would be amazing.
